pkg/client/stream_error.go

```diff
@@ -0,0 +1,44 @@
+package client
+
+import (
+	"fmt"
+	"io"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// RecvEOF should be returned when a component has returned EOF from a stream.
+// Do not use this if, for example, the initial dial to a stream fails.
+type RecvEOF struct {
+	msg string
+	err error
+}
+
+func (e *RecvEOF) Error() string {
+	return fmt.Sprintf("%s: %v", e.msg, e.err)
+}
+
+func (e *RecvEOF) Unwrap() error {
+	return e.err
+}
+
+// Is implements errors.Is for the RecvEOF type. Any error of type RecvEOF will be reported
+// as being the same as any other.
+func (e *RecvEOF) Is(target error) bool {
+	_, ok := target.(*RecvEOF)
+	return ok
+}
```

Comment:
If you use `errors.As`, then you you shouldn't need to implement an `Is` or `As` method.